### ISSUE TITLE
Parsedown Extra vs. Markdown Extra

### ISSUE THREAD

**Issue by jaswsinc** [websharks/wp-kb-articles#83]
I recently updated WPKBA so that is now uses the newer/lighter [Parsedown Extra](https://github.com/erusev/parsedown-extra) library instead of the [PHP Markdown Extra](https://github.com/michelf/php-markdown) library. I did the same in the s2Clean theme.

After some initial tests I am noticing a few subtle issues with Parsedown Extra. It seems there are a still a few kinks that need to get worked out in the Parsedown library, and in Parsedown Extra also. It's much much faster, but not doing the job yet I'm afraid.

As one quick example, _**bold** italics are broken whenever there are underscores anywhere within either_. I'm also seeing some problems with nested list parsing too. Referencing: https://github.com/erusev/parsedown/issues See also: http://parsedown.org/extra/

---

Considering the following changes:

- [x] Keep Parsedown Extra as an option, but add PHP Markdown Extra back in and set it as the default.
- [x] Add support for GitHub-style line breaks for both of these parsers, if possible.

**Comment by jaswsinc** [comment]
Keeping both in s2Clean, sticking with Parsedown Extra for WPKBA and I'm enabling support for GitHub-style line breaks, since that is expected when using GitHub integration.

---

**Original report above:** Most of this turned out to be a false alarm. The larger issue that I was having with Parsedown turned out to be a problem with WordPress itself, and it was systemic. I'll document that as part of my work on s2Clean later.

In summary:

- s2Clean now supports both PHP Markdown Extra and Parsedown Extra, with configurable options that let you choose which you prefer, and if you want GitHub-style line breaks enabled or not.
- WPKBA now uses Parsedown Extra with GitHub-style line breaks.

It's all good :smile:
